assets/components/src/plugin-settings/SettingsSection.js

```diff
@@ -27,6 +27,8 @@ const getControlComponent = setting => {
 		case 'checkbox':
 		case 'boolean':
 			return CheckboxControl;
+		case 'select':
```

Comment:
This case is not necessary. The use of `<SelectControl />` is determined by the existence of `setting.options`:

```js
	if ( Array.isArray( setting.options ) && setting.options.length ) {
		return SelectControl;
	}
```

It allows for the setting to preserve type validation.

Reply:
Thanks for clarifying! Fixed in 4eef95d6